cw/cw305/util/spiflash.py

```diff
@@ -0,0 +1,184 @@
+        'sw_strap2',
+        'tap_strap0',
+        'tap_strap1',
+    ])
+    PIN_MAPPINGS = {}
+    PIN_MAPPINGS[id(CW305)] = PinMapping(
+        sck='USB_A9',
+        sdi='USB_A10',
+        sdo='USB_A11',
+        cs='USB_A12',
+        trst='USB_A13',
+        srst='USB_A14',
+        sw_strap0='USB_A15',
+        sw_strap1='USB_A16',
+        sw_strap2='USB_A17',
+        tap_strap0='USB_A18',
+        tap_strap1='USB_A19',
+    )
+    PIN_MAPPINGS[id(CW310)] = PinMapping(
+        sck='USB_SPI_SCK',
+        sdi='USB_SPI_COPI',
+        sdo='USB_SPI_CIPO',
+        cs='USB_SPI_CS',
+        trst='USB_A13',
+        srst='USB_A14',
+        sw_strap0='USB_A15',
+        sw_strap1='USB_A16',
+        sw_strap2='USB_A17',
+        tap_strap0='USB_A18',
+        tap_strap1='USB_A19',
+    )
+    INITIAL_VALUES = PinMapping(sck=0,
+                                sdi=0,
+                                sdo=0,
+                                cs=0,
+                                trst=1,
+                                srst=1,
+                                sw_strap0=0,
+                                sw_strap1=0,
+                                sw_strap2=0,
+                                tap_strap0=0,
+                                tap_strap1=0)
+
+    RESET_DELAY = 0.1
+    BUSY_POLL_DELAY = 0.01
+    PAYLOAD_SIZE = 256
+    MAX_BUSY_ITER_CNT = 500
+
+    def __init__(self, fpga):
+        """Inits a SpiProgrammer with a CW310/305.
+
+        Args:
+          fpga: CW310/305 to be programmed, ``chipwhisperer.capture.targets.CW310/305``.
+        """
+        self.pins = self.PIN_MAPPINGS[id(type(fpga))]
+        fpga.con()
+        self.io = fpga.gpio_mode()
+        # Set strap pins as outputs
+        for pin, mapped_to in self.pins._asdict().items():
+            self.io.pin_set_output(mapped_to)
+            self.io.pin_set_state(mapped_to, getattr(self.INITIAL_VALUES, pin))
+        # Initialize SPI pins
+        self.io.spi1_setpins(sck=self.pins.sck,
+                             sdo=self.pins.sdi,
+                             sdi=self.pins.sdo,
+                             cs=self.pins.cs)
+        self.io.spi1_enable(True)
+
+    def reset(self):
+        """Resets OpenTitan."""
+        # Select JTAG and reset
+        self.io.pin_set_state(self.pins.tap_strap1, 1)
+        self.io.pin_set_state(self.pins.srst, 0)
+        time.sleep(self.RESET_DELAY)
+        # Deassert reset and return to SPI
+        self.io.pin_set_state(self.pins.srst, 1)
+        self.io.pin_set_state(self.pins.tap_strap1, 0)
+        time.sleep(self.RESET_DELAY)
+
+    def sw_strap_pins_set(self, val):
+        """Sets the state of the sw_strap pins to `val`."""
+        self.io.pin_set_state(self.pins.sw_strap0, val)
+        self.io.pin_set_state(self.pins.sw_strap1, val)
+        self.io.pin_set_state(self.pins.sw_strap2, val)
+
+    @contextmanager
+    def bootstrapper(self):
+        """Context manager for handling strap pins."""
```

Comment:
nit: It took me a bit to figure out what this does. I think it would be good to add more details to the documentation to describe that at the end of bootstrap there is an additional reset.